src/main/java/org/craftercms/engine/graphql/impl/GraphQLFactoryImpl.java

```diff
@@ -167,9 +170,11 @@ protected GraphQLSchema buildSchema(SiteContext siteContext) {
         );
 
         // Add the data fetcher for the new fields
-        codeRegistry.dataFetcher(coordinates(rootQueryTypeName, FIELD_NAME_CONTENT_ITEMS), dataFetcher);
-        codeRegistry.dataFetcher(coordinates(rootQueryTypeName, FIELD_NAME_PAGES), dataFetcher);
-        codeRegistry.dataFetcher(coordinates(rootQueryTypeName, FIELD_NAME_COMPONENTS), dataFetcher);
+        DataFetcher asyncFetcher = async(dataFetcher,
+            new SiteAwareThreadPoolExecutor(siteContext, ForkJoinPool.commonPool()));
```

Comment:
I think we can use our global job thread pool maybe?

Reply:
Right, I'll update it